service/status-report/main.go

```diff
@@ -104,6 +104,10 @@ func main() {
 		klog.Exitf("Failed to get storageClient %q/%q: %v", storageClient.Namespace, storageClient.Name, err)
 	}
 
+	if !storageClient.GetDeletionTimestamp().IsZero() {
+		klog.Exitf("StorageClient %q/%q is being deleted", storageClient.Namespace, storageClient.Name)
```

Comment:
Storage client is cluster scoped and does not have a namespace
```suggestion
		klog.Exitf("Skipping report, StorageClient %q is being deleted", storageClient.Name)
```